mlflow/genai/judges/adapters/databricks_adapter.py

```diff
@@ -0,0 +1,678 @@
+            litellm.ChatCompletionMessageToolCall(
+                id=tc["id"],
+                type=tc.get("type", "function"),
+                function=litellm.Function(
+                    name=tc["function"]["name"],
+                    arguments=tc["function"]["arguments"],
+                ),
+            )
+            for tc in tool_calls_data
+        ]
+
+    content = message_data.get("content")
+    if isinstance(content, list):
+        content_parts = [
+            block["text"] for block in content if isinstance(block, dict) and "text" in block
+        ]
+        content = "\n".join(content_parts) if content_parts else None
+
+    return litellm.Message(
+        role=message_data.get("role", "assistant"),
+        content=content,
+        tool_calls=tool_calls,
+    )
+
+
+def _serialize_messages_to_databricks_prompts(
+    messages: list["litellm.Message"],
+) -> tuple[str, str | None]:
+    """
+    Serialize litellm Messages to user_prompt and system_prompt for Databricks.
+
+    This is needed because call_chat_completions only accepts string prompts.
+
+    TODO: Replace this with a messages array parameter for call_chat_completions.
+    If call_chat_completions supported a messages parameter (similar to litellm),
+    this entire function could be removed and we could pass messages directly:
+        call_chat_completions(messages=messages, tools=tools, model=model)
+    This would make the Databricks adapter identical to the litellm adapter.
+
+    Args:
+        messages: List of litellm Message objects.
+
+    Returns:
+        Tuple of (user_prompt, system_prompt).
+    """
+    system_prompt = None
+    user_parts = []
+
+    for msg in messages:
+        if msg.role == "system":
+            system_prompt = msg.content
+        elif msg.role == "user":
+            user_parts.append(msg.content)
+        elif msg.role == "assistant":
+            if msg.tool_calls:
+                # For assistant messages with tool calls, just indicate tool usage
+                user_parts.append("Assistant: [Called tools]")
+            elif msg.content:
+                user_parts.append(f"Assistant: {msg.content}")
+        elif msg.role == "tool":
+            user_parts.append(f"Tool {msg.name}: {msg.content}")
+
+    user_prompt = "\n\n".join(user_parts)
+    return user_prompt, system_prompt
+
+
+def _run_databricks_agentic_loop(
+    messages: list["litellm.Message"],
+    trace: "Trace | None",
+    on_final_answer: Callable[[str | None], T],
+) -> T:
+    """
+    Run an agentic loop with Databricks chat completions.
+
+    This is the shared implementation for all Databricks-based agentic workflows
+    (judges, structured output extraction, etc.). It handles the iterative
+    tool-calling loop until the LLM produces a final answer.
+
+    Args:
+        messages: Initial litellm Message objects for the conversation.
+        trace: Optional trace for tool calling. If provided, enables tool use.
```

Comment:
The name of this functions seems to general, if it can only call trace-related tools